Генерація pdf

EarlGray opened the pull request:
- Генерація pdf (та дуже сирого epub) через pandoc та xelatex;
- Відправлення помилок на 0_0@гпімр.укр;

EarlGray committed:
Генерація pdf та epub.

Додано скрипт pdf/generate, який:

1) конвертує *.asc у *.xml у форматі docbook за допомогою asciidoctor;
2) конвертує docbook у сирий epub3 за допомогою pandoc;
3) конвертує docbook у LaTeX за допомогою pandoc;
4) підправляє та конвертує latex у pdf книжкового формату за допомогою
xelatex.

Залежності: pandoc, xelatex, шрифт NotoSerif.

Зауваження:
- При створенні pdf треба заміняти {stress} на наголоси;
- для збереження горизонтальних ліній на початку розділу
використовується хак: docbook має теги `<?asciidoc-hr?>` на місці `<hr>`
у вихідному asciidoc, які заміняються на буквальний текст `asciidoc-hr`,
який потрапляє у hpimr.tex, де заміняється на техівські горизонтальні
лінії (`\hrule`).

EarlGray committed:
.gitignore: pandoc/tex stuff

EarlGray committed:
прокоментовано pdf/generate

EarlGray committed:
generate-html: pdf/generate + помилки на мейл

EarlGray committed:
generate-html: повідомлення про помилку відсилання

hedrok commented on pdf/generate:
```diff
@@ -0,0 +1,41 @@
+set -e
+
+# друкувати на stderr команди, які виконуються:
+set -x
```
Круто, дякую, такого не знав.

hedrok commented on pdf/generate:
```diff
@@ -0,0 +1,41 @@
+BASEDIR=`dirname $0`       # директорія цього скрипта
+DESTDIR="../public_html"
+
+PDFTITLE="Гаррі Поттер та Методи Раціональності"
```
"та" -> "і" (так Абабагаламага переклала всі 7 книжок)

EarlGray replied:
Упс, виправив.

EarlGray committed:
Гаррі Поттер *і* методи раціональності

EarlGray committed:
pdf/generate: xelatex -interaction=nonstopmode

hedrok commented:
@EarlGray  Привіт. Раптом ти досі зацікавлений у проекті, і в тебе є час, у мене зʼявилася ідея переїхати на GitLab. Як я зрозумів, там є можливість налаштувати збірку latex. (наприклад, тут про таке https://github.com/aufenthaltsraum/stuff/wiki/Using-GitLab-CI-for-Building-LaTeX)
Але доки я зберуся це зробити ще півроку мине. Якщо ти налаштуєш все на GitLab, щоб збирався цей проект як pdf та html (опціонально - як сайт і так нормально працює), було б дуже круто.
Звісно, це все лише якщо в тебе є бажання й час.

olgabyelova committed:
перекладено не перечитано

hedrok committed:
Зауваження від анонімного читача

hedrok committed:
Анонімні зауваження

hedrok committed:
Зауваження від asdflok2

hedrok committed:
Зауваження від Романа

hedrok committed:
Зауваження від читача Володимира

hedrok committed:
Анонімні зауваження

hedrok committed:
Зауваження від читача Володимира

hedrok committed:
Зауваження від asdflok2

hedrok committed:
Зауваження від анонімних читачів

hedrok committed:
Розділ 114: вичитування

hedrok committed:
Розділ 116: вичитування

hedrok committed:
Розділ 116

brownian committed:
052.asc: виправлено після обговорення

hedrok committed:
!!!

olgabyelova committed:
розділ 117 переклад без вичитки

hedrok committed:
Розділ 115: переклад

hedrok committed:
Розділ 114

hedrok committed:
Розділ 115: вичитування

hedrok committed:
Розділи 117, 118: вичитування

hedrok committed:
Розділ 116: зауваження від читача Володимира

hedrok committed:
Розділи 114-118: публікація

hedrok committed:
Анонімні зауваження

hedrok committed:
Розділ 119: назва й зірки

hedrok committed:
Розділ 119: відсотки

hedrok committed:
Розділ 119: 14%

brownian committed:
056.asc: вичитане

hedrok committed:
Розділ 119: 56%

hedrok committed:
Розділ 119: 70%

hedrok committed:
Розділ 119: 100%

hedrok committed:
Розділ 119: 70% вичитування

hedrok committed:
Розділ 119: вичитування

ElviraIzmailova committed:
переклад 120 - бувай, Драко Т_Т

hedrok committed:
Розділ 120: вичитування

hedrok committed:
Розділ 120: після обговорення

hedrok committed:
Зауваження від читачів

hedrok committed:
Merge branch 'ch120_review' into 'master'

Розділ 120: вичитування

See merge request hpimr/hpimr!439

hedrok committed:
Розділ 119: зауваження читача Володимира

hedrok committed:
Розділ 119

hedrok committed:
Розділи 119, 120: публікація

hedrok committed:
Розділ 104: зауваження від Романа

hedrok committed:
Розділ 120: held court - хай поки буде двір

brownian committed:
054.asc: змій, змі{stress}я

hedrok committed:
Merge branch 'ch054' into 'master'

054.asc: вичитане

See merge request hpimr/hpimr!434

hedrok committed:
Merge branch 'ch052' into 'master'

052.asc: вичитане

See merge request hpimr/hpimr!427

brownian committed:
056.asc: після обговорення

brownian committed:
057.asc: вичитане

brownian committed:
056.asc: «чи»

brownian committed:
057.asc: __загальне__ уявлення про їхню… → *кожного*

hedrok committed:
Merge branch 'ch056' into 'master'

056.asc: вичитане

See merge request hpimr/hpimr!438

brownian committed:
057.asc: "а не лише той" -> "а не лише того"

hedrok committed:
Merge branch 'ch057' into 'master'

057.asc: вичитане

See merge request hpimr/hpimr!440

hedrok committed:
Розділ 121: вичитування

olgabyelova committed:
майже все

olgabyelova committed:
все

hedrok committed:
Розділ 122: вичитування

hedrok committed:
Розділи 121, 122: публікація

hedrok committed:
Merge branch 'ch121-review'

hedrok committed:
Розділ 121: технічне виправлення

brownian committed:
058.asc: вичитане

brownian committed:
П'ять розділів з пробілами після «ч.» перед цифрою в назві

brownian committed:
059.asc: вичитане

brownian committed:
060.asc: вичитане

hedrok committed:
Merge branch 'ch058' into 'master'

058.asc: вичитане

See merge request hpimr/hpimr!441

hedrok committed:
Merge branch 'ch_no_space' into 'master'

П'ять розділів з пробілами після «ч.» перед цифрою в назві

See merge request hpimr/hpimr!442

hedrok committed:
Fix #390 Плащ Невидимості -> Плащ невидимості

hedrok committed:
Зауваження від читачів

brownian committed:
055.asc: про відчуття безнадії

brownian committed:
060.asc: повернув «Гадаю, це відповідає…»

brownian committed:
061.asc: вичитане

brownian committed:
062.asc: вичитане

hedrok committed:
Зауваження від читачів

brownian committed:
063.asc: вичитане

brownian committed:
глузд → ґлузд, як у більшості розділів.

brownian committed:
065.asc: вичитане

brownian committed:
066.asc: вичитане

brownian committed:
067.asc: вичитане

hedrok committed:
Merge branch 'ch059' into 'master'

059.asc: вичитане

See merge request hpimr/hpimr!443

hedrok committed:
Merge branch 'ch060' into 'master'

060.asc: вичитане

See merge request hpimr/hpimr!444

hedrok committed:
Merge branch 'ch061' into 'master'

061.asc: вичитане

See merge request hpimr/hpimr!445

hedrok committed:
Merge branch 'ch062' into 'master'

062.asc: вичитане

See merge request hpimr/hpimr!446

hedrok committed:
Merge branch 'gluzd' into 'master'

глузд → ґлузд, як у більшості розділів.

See merge request hpimr/hpimr!448

hedrok committed:
Merge branch 'ch067' into 'master'

067.asc: вичитане

See merge request hpimr/hpimr!451

brownian committed:
055.asc: "вони" -> "воно"

brownian committed:
063.asc після обговорення

brownian committed:
065.asc: після обговорення

brownian committed:
066.asc: після обговорення

brownian committed:
068.asc: вичитане

brownian committed:
069.asc: вичитане; 072.asc: стейкА

brownian committed:
070.asc: вичитане

brownian committed:
071.asc: вичитане

brownian committed:
072.asc: вичитане

brownian committed:
073.asc: вичитане

brownian committed:
074.asc: вичитане

brownian committed:
075.asc: вичитане

brownian committed:
072.asc: перепрошую, відкотив виділення прямої мови

brownian committed:
076.asc: вичитане

brownian committed:
077.asc: вичитане

brownian committed:
078.asc: вичитане (плюс правка у кількох інших)

brownian committed:
079.asc: якось вичитане

brownian committed:
080.asc: вичитане

brownian committed:
081.asc: вичитане

brownian committed:
082.asc: вичитане

brownian committed:
083.asc: вичитане

brownian committed:
084.asc: вичитане

brownian committed:
085.asc: вичитане якось

brownian committed:
086.asc: вичитане

brownian committed:
087.asc: вичитане

brownian committed:
088.asc: вичитане

brownian committed:
089.asc: вичитане

brownian committed:
090.asc: вичитане

brownian committed:
091.asc: вичитане

brownian committed:
092.asc: вичитане

brownian committed:
093.asc: вичитане

brownian committed:
094.asc: вичитане

brownian committed:
095.asc: вичитане

brownian committed:
096.asc: вичитане

brownian committed:
097.asc: вичитане

brownian committed:
098.asc: вичитане

brownian committed:
100.asc: вичитане

brownian committed:
101.asc: вичитане

brownian committed:
102.asc: вичитане

brownian committed:
103.asc: вичитане

brownian committed:
088.asc: МерліновІ (присвійний)

brownian committed:
104.asc: вичитане

brownian committed:
105.asc: вичитане

brownian committed:
106.asc: вичитане

brownian committed:
107.asc: вичитане

brownian committed:
108.asc: вичитане

brownian committed:
078.asc: рецептА

brownian committed:
109.asc: вичитане

brownian committed:
110.asc: вичитане

brownian committed:
111.asc: вичитане

brownian committed:
index.asc: помилка для 113-го

hedrok committed:
Merge branch 'index' into 'master'

index.asc: помилка для 113-го

See merge request hpimr/hpimr!495

brownian committed:
112.asc: вичитане

brownian committed:
113.asc: вичитане

brownian committed:
113.asc: 2015-го року

brownian committed:
114.asc: вичитане

brownian committed:
115.asc: вичитане

brownian committed:
116.asc: вичитане

brownian committed:
117.asc: вичитане

brownian committed:
118.asc: вичитане

hedrok committed:
Merge branch 'ch055' into 'master'

055.asc: вичитане

See merge request hpimr/hpimr!435

hedrok committed:
Merge branch 'ch063' into 'master'

063.asc: вичитане

See merge request hpimr/hpimr!447

brownian committed:
119.asc: вичитане

brownian committed:
120.asc: вичитане

brownian committed:
121.asc: вичитане

hedrok committed:
Merge branch 'ch121' into 'master'

121.asc: вичитане

See merge request hpimr/hpimr!505

brownian committed:
122.asc: вичитане

hedrok committed:
Merge branch 'ch065' into 'master'

065.asc: вичитане

See merge request hpimr/hpimr!449

hedrok committed:
Парселмова в 65-му та 66-ому

hedrok committed:
Merge branch 'ch66-review'

hedrok committed:
Merge branch 'ch68-review'

hedrok committed:
Merge branch 'ch69-review'

hedrok committed:
Merge branch 'ch70-review'

hedrok committed:
Розділ 70: одрук + зловісний сміх

brownian committed:
119.asc: пеан… відлунював

hedrok committed:
Merge branch 'ch71-review'

hedrok committed:
Розділ 072: чотири дівчини -> чотири дівчата

hedrok committed:
Merge branch 'ch72-review'

hedrok committed:
Розділ 73: вичитування вичитування

* знизав -> стенув: повертаю. СУМ вказує рідко, але знає таке.
  У нас 4 "стенув" на 50 "знизав.
* "зарізким. -- Саме" -> "зарізким, -- саме" в оригіналі з маленької
* "кістки з адамантовою арматурою" -> "покриті адамантієм кістки"
  Прозорий натяк на Росомаху. Згідно вікі, в нього кістки покриті
  адамантієм.

hedrok committed:
Merge branch 'ch73-review'

hedrok committed:
Розділ 74: вичитування вичитування

Як на мене, коли Квірел незадоволений Гаррі, він розмовляє на "ви".

hedrok committed:
Merge branch 'ch74-review'

hedrok committed:
Merge branch 'ch75-review'

hedrok committed:
Merge branch 'ch76-review'

hedrok committed:
Merge branch 'ch77-review'

hedrok committed:
Повернув "з такими-то шансами"

hedrok committed:
Розділ 78: вичитування

hedrok committed:
Повернув карту: вона в каноні Карта мародера

hedrok committed:
Розділ 79: вичитування

hedrok committed:
Розділ 80: коректура

hedrok committed:
Розділ 81: коректура

hedrok committed:
Розділ 82: передав yea як ага, nay -> а-а

hedrok committed:
Розділ 82: коректура

Commit:
Виправлення - видалення зайвого слова <стукнула>

hedrok committed:
Merge branch 'master' into 'master'

Виправлення - видалення зайвого слова <стукнула>

See merge request hpimr/hpimr!507

hedrok committed:
Розділ 83: узгодження

hedrok committed:
Розділ 83: коректура

hedrok committed:
Розділ 84: вичитування вичитування

* Додав тире
* Про "одержимість": йдеться про те, що Волдеморт може захоплювати тіла інших людей - як Джіні в каноні. Переформулював повністю.
* Переформулював про жест

hedrok committed:
Розділ 84: коректура

hedrok committed:
Оновлення master

hedrok committed:
Розділ 85: коректура коректури

And so it was done:
"І, отже, це сталося" -- мені здається, що так складається враження, що
саме на це Дамблдор і чекав. З оригіналу, та й далі по тексту видно, що
Дамблдор сам не знав, як краще. Як на мене, тут треба передати, що для
самого Дамблдора це нова інформація -- **як** сталося. Тому повертаю
свій варіант "І так воно сталося".

hedrok committed:
Розділ 85: коректура

hedrok committed:
Розділ 86: вичитування коректури

* Як пророцтво звучало, коли було зробленО

Мені здається, тут саме зроблено. Бо зробленЕ воно й зараз.

* Посеред траєкторії/посеред повітря

Оригінал "in the midair". Якщо поставити "посеред траєкторії", то можна
сприйняти як напівшляху. Трохи переформулював.

* Пил/жар емоцій

Пошукав, не знайшов жодного з цих варіантів. Виправив на "розпал емоцій"

* І всілякі дрібниці

hedrok committed:
Розділ 86: коректура

hedrok committed:
Розділ 87: коректура

hedrok committed:
Розділ 88: коректура

hedrok committed:
Розділ 89: коректура

hedrok committed:
Розділ 90: коректура

hedrok committed:
Розділ 91: коректура

hedrok committed:
Розділ 92: коректура

hedrok committed:
Розділ 93: вичитування коректури

hedrok committed:
Розділ 93: коректура

hedrok committed:
Розділ 94: коректура

hedrok committed:
Розділ 95: коректура

hedrok committed:
Розділ 96: вичитування коректури

* Додав пропущене - дуже дякую, що нагадав
* Перевірив ще раз канон про The last enemy - там ще "це" в перекладі.
  Кому залишив, хоча її в перекладі канону немає.
* Повернув "моя мама й мій тато" -- в оригіналі це підкреслено, вважаю,
  що варто зберегти.

hedrok committed:
Розділ 96: коректура

hedrok committed:
Розділ 97: вичитування коректури

* Прибрав кому

hedrok committed:
Розділ 97: коректура

hedrok committed:
Розділ 98: коректура

hedrok committed:
Розділ 100: коректура

hedrok committed:
Розділ 101: коректура

hedrok committed:
Розділ 102: ще трохи парселмови

hedrok committed:
Розділ 102: коректура

hedrok committed:
Розділ 103: коректура

hedrok committed:
Розділ 104: назва

hedrok committed:
Розділ 104: коректура

hedrok committed:
Розділ 105: коректура

hedrok committed:
Розділ 108: вичитування коректури

hedrok committed:
Розділ 106: коректура

hedrok committed:
Розділ 107: коректура

hedrok committed:
Розділ 108: коректура

hedrok committed:
Розділ 109: коректура

hedrok committed:
Розділ 110: коректура

hedrok committed:
Розділ 111: коректура

hedrok committed:
Розділ 112: коректура

EarlGray committed:
Full book pdf; a nicer front page

hedrok committed:
Розділ 113: додав форматування

hedrok committed:
Розділ 113: коректура

hedrok committed:
Розділ 114: трохи переформулював

hedrok committed:
Розділ 114: коректура

hedrok committed:
Розділ 115: коректура

hedrok committed:
Розділ 116: тривалість матчів

hedrok committed:
Розділ 116: коректура

hedrok committed:
Розділ 117: пунктуація, узгодження, посада

hedrok committed:
Розділ 117: коректура

hedrok committed:
Розділ 118: коректура

hedrok committed:
Розділ 119: дрібні виправлення

hedrok committed:
Розділ 119: коректура

hedrok committed:
Розділ 120: дрібниці

hedrok committed:
Розділ 120: коректура

hedrok committed:
Розділ 122: а ля

hedrok committed:
Розділ 122: коректура

hedrok committed:
Зауваження майже за рік

hedrok committed:
Додаткові виправлення

hedrok committed:
Зауваження Олександра Пилипчука

hedrok committed:
fix #391

hedrok committed:
Пропозиції Юлії Єловської

hedrok committed:
fix #391 тепер правильно

int-ua committed:
Виправлення з LanguageTool +

hedrok committed:
Merge branch 'languagetool' into 'master'

Виправлення з LanguageTool +

See merge request hpimr/hpimr!510
